rest/route/repo.go

```diff
@@ -57,9 +66,320 @@ func (h *repoIDGetHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	repoModel.Variables = *repoVars
 
-	if repoModel.Aliases, err = h.sc.FindProjectAliases("", repo.Id); err != nil {
+	if repoModel.Aliases, err = h.sc.FindProjectAliases("", repo.Id, nil); err != nil {
 		return gimlet.MakeJSONErrorResponder(err)
 	}
 
 	return gimlet.NewJSONResponse(repoModel)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/repos/{repo_id}
+
+type repoIDPatchHandler struct {
+	repoName       string
+	originalRepo   *dbModel.RepoRef
+	newRepoRef     *dbModel.RepoRef
+	requestRepoRef *model.APIProjectRef
+	user           *user.DBUser
+
+	sc       data.Connector
+	settings *evergreen.Settings
+}
+
+func makePatchRepoByID(sc data.Connector, settings *evergreen.Settings) gimlet.RouteHandler {
+	return &repoIDPatchHandler{
+		sc:       sc,
+		settings: settings,
+	}
+}
+
+func (h *repoIDPatchHandler) Factory() gimlet.RouteHandler {
+	return &repoIDPatchHandler{
+		sc:       h.sc,
+		settings: h.settings,
+	}
+}
+
+func (h *repoIDPatchHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.repoName = gimlet.GetVars(r)["repo_id"]
+	h.user = MustHaveUser(ctx)
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+
+	// get the old repo
+	h.originalRepo, err = dbModel.FindOneRepoRef(h.repoName)
+	if err != nil {
+		return err
+	}
+	if h.originalRepo == nil {
+		return errors.Errorf("repo '%s' doesn't exist", h.repoName)
+	}
+
+	h.requestRepoRef = &model.APIProjectRef{}
+	if err = h.requestRepoRef.BuildFromService(h.originalRepo.ProjectRef); err != nil {
+		return errors.Wrap(err, "API error converting from model.ProjectRef to model.APIProjectRef")
+	}
+
+	// erase contents so requestRepoRef will only be populated with new elements for these fields
+	h.requestRepoRef.Admins = nil
+	h.requestRepoRef.GitTagAuthorizedUsers = nil
+	h.requestRepoRef.GitTagAuthorizedTeams = nil
+	if err = json.Unmarshal(b, h.requestRepoRef); err != nil {
```

Comment:
Are you sure that unmarshalling into an already-populated struct is safe to do? I haven't tried it but I'm suspicious that it might overwrite values in unexpected ways

Reply:
This is what we do for the modify project ref route too, and it does work; it is a little strange, but if we don't do this, then for _every_ setting we have to check and see if it's been modified from the original, or we could switch to a route that works like the modify distro route (PUT, not PATCH), but then i'd also want to depreciate the existing project_ref PATCH route, and create a new PUT route. (This would be more upfront work but honestly not the worst thing)